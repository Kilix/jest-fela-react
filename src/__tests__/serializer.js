import React from 'react';
import { createComponent, Provider } from 'react-fela';
import { createRenderer } from 'fela';
import renderer from 'react-test-renderer';
import monolithic from 'fela-monolithic';

import createSerializer from '../serializer';

const Text = createComponent(() => ({ color: 'red', fontSize: '16px' }));
const Wrapper = createComponent(() => ({
    border: 'solid 1px black',
    '> div': {
        fontWeight: 500,
    },
}));

const felaRenderer = createRenderer({ enhancers: [monolithic()] });
expect.addSnapshotSerializer(createSerializer(felaRenderer));

it('react-test-renderer', () => {
    const tree = renderer
        .create(
            <Provider renderer={felaRenderer}>
                <Wrapper>
                    <Text>Hello World</Text>
                </Wrapper>
            </Provider>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
