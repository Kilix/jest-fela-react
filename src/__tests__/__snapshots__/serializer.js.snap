// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer 1`] = `
.fovk07z> div {
  font-weight: 500;
}

.fovk07z {
  border: solid 1px black;
}

.f16lnj6g {
  color: red;
  font-size: 16px;
}

<div
  className="fovk07z"
>
  <div
    className="f16lnj6g"
  >
    Hello World
  </div>
</div>
`;
